### PR TITLE
Add AI_FLAG_OMNISCIENT

### PR DESCRIPTION
**Pull request by ghoulslash**
Creates an AI_FLAG_OMNISCIENT ai flag where trainer knows your full party's moves, abilities, and hold items.

### CODE REVIEW

**Review by LOuroboros — CHANGES_REQUESTED** [review]

File: src/battle_main.c [hunk 1]
```diff
@@ -4096,7 +4096,8 @@ static void HandleTurnActionSelectionState(void)
 
             // Do AI score computations here so we can use them in AI_TrySwitchOrUseItem
             if ((gBattleTypeFlags & BATTLE_TYPE_HAS_AI || IsWildMonSmart())
-                && (IsBattlerAIControlled(gActiveBattler) && !(gBattleTypeFlags & BATTLE_TYPE_PALACE))) {
+                    && (BattlerHasAi(gActiveBattler) && !(gBattleTypeFlags & BATTLE_TYPE_PALACE)))
```
Comment (LOuroboros) [inline src/battle_main.c:4099]: The suggestion was to stack the conditions  so the 2nd line's checks could be read right below the `if` statement and the checks in its same line. You only aded even more spacing which is not what I suggested 😂 ...

```c
            if ((gBattleTypeFlags & BATTLE_TYPE_HAS_AI || IsWildMonSmart())
             && (BattlerHasAi(gActiveBattler) && !(gBattleTypeFlags & BATTLE_TYPE_PALACE)))
```

Comment (ghoulslash) [reply src/battle_main.c:4099]: As per discord discussion it doesn't seem like there is a universal formatting for things like this so I would prefer to leave it as is

File: src/battle_ai_util.c [hunk 2]
```diff
@@ -473,6 +473,14 @@ bool32 IsBattlerAIControlled(u32 battlerId)
     }
 }
 
+bool32 IsBattlerAIControlled(u32 battlerId)
```
Comment (AsparagusEduardo) [inline src/battle_ai_util.c:476]: Wouldn't this cause issues with how AI detects if a Pokémon is not controlled by a player?

Comment (ghoulslash) [reply src/battle_ai_util.c:476]: It's not used in that many places. And the relevant places have been replaced with BattlerHasAi

Comment (AsparagusEduardo) [reply src/battle_ai_util.c:476]: It's better now, but the name doesn't sit well for me.
If I call this function with a player-controlled battlerId and the omniscient flag is on, the function name should conceptually still return FALSE do to that battler not being player-controlled, but it's not doing that currently.

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**